### PR TITLE
Fixes and addon for BERT usage

### PR DESCRIPTION
**Pull request by OfirArviv**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,26 @@ Run the parser on a French/German text file (separate passages by blank lines):
...
+To use them pass the `--use-bert` argument in the relevant command and install the packages in  requirements.bert.txt:
+
+
+    python -m pip install -r requirements.bert.txt
+
+See the possible config options in `config.py` (relevant configs are with the prefix `bert`).
+
+### Using BERT embeddings: Multilingual training
+It's possible, when using the BERT embeddings, to train a multilingual model which can leverage
+cross-lingual transfer and improve results on low-show languages.
```
Comment (danielhers) [inline README.md]: What's "low-show"? Maybe "low-resource"?

Comment (OfirArviv) [reply README.md]: shot, a typo

File: tupa/parse.py [hunk 2]
```diff
@@ -644,6 +644,28 @@ def read_passages(args, files):
                                       converters=CONVERTERS, lang=Config().args.lang)
 
 
+def filter_passages_for_bert(passages, args):
```
Comment (danielhers) [inline tupa/parse.py]: Shouldn't this return a generator? In case we don't want to load all files to memory before starting to train/parse.

Comment (OfirArviv) [reply tupa/parse.py]: Generator doesn't work with the shuffle function.
I agree its not optimal and should maybe be fixed in the future (or now if you have a better idea :) )

**Comment by danielhers** [comment]
Thanks. See also the changes I made to neural_network.py in the mrp branch: https://github.com/danielhers/tupa/tree/mrp
Do we want to incorporate them too?

File: tupa/parse.py (hunk 2 above)
Comment (danielhers) [reply tupa/parse.py]: Oh, with a [LazyLoadedPassages](https://github.com/danielhers/ucca/blob/master/ucca/ioutil.py#L21) object the shuffle will shuffle the filenames and still the loading of the actual passages is lazy.

**Comment by OfirArviv** [comment]
> 
> 
> Thanks. See also the changes I made to neural_network.py in the mrp branch: https://github.com/danielhers/tupa/tree/mrp
> Do we want to incorporate them too?

The error messages: Yes, I think we do.

File: tupa/parse.py (hunk 2 above)
Comment (OfirArviv) [reply tupa/parse.py]: Can you confirm that doing the filtering in Parser.train() and Parser.parse() will cover all the code paths?

Comment (danielhers) [reply tupa/parse.py]: I think it should be done together with `annotate_all`. Then it will cover all paths.

Comment (OfirArviv) [reply tupa/parse.py]: annotate_all is done only in Parser.parse() function. I will do it therre

Comment (danielhers) [reply tupa/parse.py]: Right. Since train() calls parse(), it will work.

Comment (danielhers) [reply tupa/parse.py]: Great, now just need to yield instead of return a list :)

Comment (OfirArviv) [reply tupa/parse.py]: Oh right, forgot about it ^^' Good tihng you are taking a second look

**Pull request merged**